flags.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type kwargStringFlag struct {
+	value   *map[string][]byte
+	changed bool
+}
+
+func newKwargStringFlag() *kwargStringFlag {
+	return &kwargStringFlag{
+		value: new(map[string][]byte),
+	}
+}
+
+func (f *kwargStringFlag) AsMap() map[string][]byte {
+	var m map[string][]byte
+	if f.value != nil {
+		return *f.value
+	}
+	return m
+}
+
+func (f *kwargStringFlag) String() string {
+	m := f.AsMap()
+	return fmt.Sprintf("%v", m)
+}
+
+func (f *kwargStringFlag) Set(input string) error {
+	p, err := parseKeyValuePair(input)
+	if err != nil {
+		return err
+	}
+	if !f.changed {
+		// first time set is called so assign to *f.value
+		*f.value = map[string][]byte{p.Key: p.Value}
+	} else {
+		// after first time just update existing *f.value
+		(*f.value)[p.Key] = p.Value
+	}
+	f.changed = true
+	return nil
+}
+
+func (f *kwargStringFlag) Type() string {
+	return "key=value"
+}
+
+type pair struct {
+	Key   string
+	Value []byte
+}
+
+func parseKeyValuePair(input string) (*pair, error) {
+	if input == "" {
+		return nil, nil
+	}
+	pairSplit := bytes.SplitN([]byte(input), []byte("="), 2)
+	if len(pairSplit) == 1 {
+		return nil, fmt.Errorf("did not find any key=value pairs in %s)", input)
+	}
+	key := string(pairSplit[0])
+	value := pairSplit[1]
+	return &pair{key, value}, nil
+}
+
+type positionalArgStringFlag struct {
+	value   *[][]byte
```

Comment:
Why use a pointer when you can just use nil?

Also, why use []byte over string. It seems like we're doing conversions to strings whenever we actually want to use these values.

Reply:
This is a pattern to allow knowing if the value has ever been set before. I copied it from the pflag stringSlice code. I guess for slices, it might not be necessarily since you can append to an empty slice. I'm not entirely sure why they didn't do it that way. I can try to change that.

I'm using a byte slice because we're pretty much just unmarshalling these values from JSON which needs `[]byte`.

Reply:
I removed the usage of the pointer here, and am still currently using `[][]byte`